zeebe/exporters/rdbms-exporter/src/main/java/io/camunda/exporter/rdbms/DecisionInstanceExportHandler.java

```diff
@@ -0,0 +1,125 @@
+import io.camunda.zeebe.protocol.record.value.EvaluatedInputValue;
+import io.camunda.zeebe.protocol.record.value.MatchedRuleValue;
+import io.camunda.zeebe.util.DateUtil;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DecisionInstanceExportHandler
+    implements RdbmsExportHandler<DecisionEvaluationRecordValue> {
+
+  private final DecisionInstanceWriter decisionInstanceWriter;
+
+  public DecisionInstanceExportHandler(final DecisionInstanceWriter decisionInstanceWriter) {
+    this.decisionInstanceWriter = decisionInstanceWriter;
+  }
+
+  @Override
+  public boolean canExport(final Record<DecisionEvaluationRecordValue> record) {
+    return record.getValueType() == ValueType.DECISION_EVALUATION
+        && record.getIntent() == DecisionEvaluationIntent.EVALUATED;
+  }
+
+  @Override
+  public void export(final Record<DecisionEvaluationRecordValue> record) {
+    final DecisionEvaluationRecordValue value = record.getValue();
+
+    int index = 0;
+    for (final EvaluatedDecisionValue evaluatedDecision : value.getEvaluatedDecisions()) {
+      final var state = getState(record, value, index);
+      final var key = record.getKey() + index; // TODO fix to real key?
+      final var decisionInstance =
+          new DecisionInstanceDbModel.Builder()
+              .decisionInstanceKey(key)
+              .decisionDefinitionKey(evaluatedDecision.getDecisionKey())
+              .decisionDefinitionId(evaluatedDecision.getDecisionId())
+              .evaluationDate(
+                  DateUtil.toOffsetDateTime(Instant.ofEpochMilli(record.getTimestamp())))
+              .processDefinitionKey(value.getProcessDefinitionKey())
+              .processDefinitionId(value.getBpmnProcessId())
+              .processInstanceKey(value.getProcessInstanceKey())
+              .decisionRequirementsKey(value.getDecisionRequirementsKey())
+              .decisionRequirementsId(value.getDecisionRequirementsId())
+              .flowNodeInstanceKey(value.getElementInstanceKey())
+              .flowNodeId(value.getElementId())
+              .rootDecisionDefinitionKey(evaluatedDecision.getDecisionKey())
+              .result(evaluatedDecision.getDecisionOutput())
+              .evaluatedInputs(createEvaluationInputs(key, evaluatedDecision.getEvaluatedInputs()))
+              .evaluatedOutputs(createEvaluationOutputs(key, evaluatedDecision.getMatchedRules()))
+              .state(state)
+              .decisionType(DecisionDefinitionType.fromValue(evaluatedDecision.getDecisionType()))
+              .evaluationFailure(
+                  state == DecisionInstanceState.FAILED
+                      ? value.getEvaluationFailureMessage()
+                      : null)
+              .build();
+
+      decisionInstanceWriter.create(decisionInstance);
+      index++;
+    }
+  }
+
+  // TODO move to common exporter util module
```

Comment:
Do we have an Issue for this?

Reply:
[#24002](https://github.com/camunda/camunda/issues/24002)